browser/onboarding/onboarding_unittest.cc

```diff
@@ -107,7 +108,10 @@ TEST_F(OnboardingTest, HelperCreationTestForNonFirstRun) {
   TestingBrowserProcess::GetGlobal()->local_state()->SetTime(
       onboarding::prefs::kLastShieldsIconHighlightTime, base::Time());
   OnboardingTabHelper::MaybeCreateForWebContents(web_contents.get());
-  tab_helper = OnboardingTabHelper::FromWebContents(web_contents.get());
+  ASSERT_TRUE(base::test::RunUntil([&tab_helper, &web_contents]() {
```

Comment:
This change works in this case but I think we need to improve all other tests in this file.
Some tests expect web contents is not created from the below reply callback but we don't wait it now.
I could f/l this later.
```
  base::ThreadPool::PostTaskAndReplyWithResult(
      FROM_HERE, {base::MayBlock()},
      base::BindOnce(&OnboardingTabHelper::IsSevenDaysPassedSinceFirstRun),
      base::BindOnce(
          [](base::WeakPtr<content::WebContents> contents, bool passed) {
            if (!passed && contents) {
              OnboardingTabHelper::CreateForWebContents(contents.get());
            }
          },
          web_contents->GetWeakPtr()));
```

Reply:
I can use `base::test::RunUntil` to wait for the result to become `true`, but I can't just wait for the result to never change. So it would require some further rework to be able to to test for the false cases, and that may very well require you to decide how to change this code.